icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -5037,28 +5035,24 @@ void RBBITest::TestTraceCreateBreakEngine(void) {
     // Loop through all the text.
     while (brkitr->next() > 0) ;
 
-    assertEquals("utrace_entry should be called ", 6, gEntryFn.size());
-    assertEquals("utrace_exit should be called ", 6, gExitFn.size());
-    assertEquals("utrace_data should be called ", 5, gDataFn.size());
+    std::ostringstream stringStream;
 
-    assertEquals("utrace_entry should be called ", UTRACE_UBRK_CREATE_WORD, gEntryFn[0]);
-    assertEquals("utrace_exit should be called ", UTRACE_UBRK_CREATE_WORD, gExitFn[0]);
+    stringStream << "utrace_entry got called " << gEntryFn.size() << " times\n"
```

Comment:
When someone in the future tries to read this test, it won't be clear what was the expected result.  They'll get a bunch of output, but it's not clear what the output was supposed to be.

Suggestion: How about making a little function `assertEqualsOptional` that uses logln instead of errln when the arguments don't match?  Then your code is still readable.  You could put the function either here or in intltest.cpp.

Reply:
It will be confusing to setup expectation without calling a clean up in front. I now hook up to the rbbi clean up only from test and see will that help.